scheduler/task.go

```diff
@@ -353,6 +354,16 @@ func (t *task) Stop() {
 	defer t.Unlock()
 	if t.state == core.TaskFiring || t.state == core.TaskSpinning {
 		t.state = core.TaskStopping
+		for {
+			_, ok := <-t.isFiring
```

Comment:
What is this waiting? It could wait forever. I don't see anywhere single t.isFiring is done.